WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java

```diff
@@ -80,366 +26,45 @@ public void onCreate(Bundle savedInstanceState) {
-                && changeType == CommentActions.ChangeType.EDITED) {
-            FragmentManager fm = getFragmentManager();
-            if (fm.getBackStackEntryCount() > 0) {
-                fm.popBackStack();
-            }
-        }
-    }
-
-    private class NoteClickListener implements NotificationsListFragment.OnNoteClickListener {
-        @Override
-        public void onClickNote(String noteId) {
-            if (TextUtils.isEmpty(noteId)) return;
-
-            // open the latest version of this note just in case it has changed - this can
-            // happen if the note was tapped from the list fragment after it was updated
-            // by another fragment (such as NotificationCommentLikeFragment)
-            if (SimperiumUtils.getNotesBucket() != null) {
-                try {
-                    Note note = SimperiumUtils.getNotesBucket().get(noteId);
-                    openNote(note);
-                } catch (BucketObjectMissingException e) {
-                    AppLog.e(T.NOTIFS, "Note could not be found.");
-                }
-            }
-        }
-    }
-
-    // Retrieves comment reply text so we can restore it upon returning to CommentDetailFragment
-    private String getCommentReplyText() {
-        if (mDetailFragment != null && mDetailFragment instanceof CommentDetailFragment) {
-            CommentDetailFragment commentDetailFragment = (CommentDetailFragment)mDetailFragment;
-            if (!TextUtils.isEmpty(commentDetailFragment.getReplyText())) {
-                return commentDetailFragment.getReplyText();
-            }
+        Intent intent = getIntent();
+        if (intent.hasExtra(NotificationsListFragment.NOTE_ID_EXTRA)) {
+            boolean shouldShowKeyboard = intent.getBooleanExtra(NotificationsListFragment.NOTE_INSTANT_REPLY_EXTRA, false);
+            mNotesListFragment.openNote(intent.getStringExtra(NotificationsListFragment.NOTE_ID_EXTRA), this, shouldShowKeyboard);
         }
 
-        return null;
+        GCMIntentService.clearNotificationsMap();
     }
 
     @Override
-    public void onSaveInstanceState(@Nonnull Bundle outState) {
-        if (outState.isEmpty()) {
-            outState.putBoolean("bug_19917_fix", true);
-        }
-        outState.putBoolean(KEY_INITIAL_UPDATE, mHasPerformedInitialUpdate);
-        if (getFragmentManager().getBackStackEntryCount() > 0
-                || getFragmentManager().findFragmentByTag(TAG_TABLET_DETAIL_VIEW) != null) {
-            outState.putString(NOTE_ID_EXTRA, mSelectedNoteId);
-        }
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
 
-        // Save text in comment reply EditText
-        if (!TextUtils.isEmpty(getCommentReplyText())) {
-            outState.putString(KEY_REPLY_TEXT, getCommentReplyText());
-        }
-
-        // Save list view scroll position
+        // This can be removed if min_sdk is 16 or higher
```

Comment:
I'm using `startActivityForResult` from a fragment and I want to get the `onActivityResult` there as well. 

`startActivityForResult` wasn't added to `Fragment` until api 16 so I added this workaround.